quarkus/tests/junit5/src/main/java/org/keycloak/it/junit5/extension/CLITestExtension.java

```diff
@@ -177,12 +177,11 @@ private void onKeepServerAlive(KeepServerAlive annotation) {
     public void afterEach(ExtensionContext context) throws Exception {
         DistributionTest distConfig = getDistributionConfig(context);
 
-        if (distConfig != null) {
-            if (distConfig.keepAlive()) {
-                dist.stop();
-            }
+        if (dist != null) {
+            onKeepServerAlive(context.getRequiredTestMethod().getAnnotation(KeepServerAlive.class), false);
+            dist.stop();
```

Comment:
Technically, we might be stopping already stopped server here. But that should be ok.

Reply:
Right I see the close operations as being tolerant to multiple close calls.